### ISSUE TITLE
How to pass custom error message when doing async submit?

### ISSUE THREAD

**Issue by kid** [davidkpiano/react-redux-form#24]
Right now the errors seems to only boolean, is there anyway to pass custom error message to the form reducer? 
For example I'm using Auth0 for authentication and I would like to display error message returned by Auth0 itself.

**Comment by davidkpiano** [comment]
I'm actually working on integrating this feature (for use of any error messages and/or complex values to be used for both validation and errors), but I'm wondering, what is the shape of the data for the custom error messages that you receive from Auth0?

**Comment by davidkpiano** [comment]
So here are the planned ideas I have for this:

A new `actions.setErrors(model, errors)` action creator that acts like `setValidity`, except it can set values of any type directly to the `errors` property of the field:

```js
dispatch(actions.setErrors('user.email', {
  length: 'Your email is not long enough',
  illegal: 'Your email contains illegal characters.'
});

getField(userForm, 'email').errors;
// => { length: 'Your email is not long enough', illegal: 'Your email contains illegal characters' }

getField(userForm, 'email').validity;
// => { length: false, illegal: false }
```

And a new `actions.submit(model, promise)` action thunk creator that takes a promise and does the following:

- If promise is resolved, `setValidity()` to the resolved value.
- If promise is rejected, `setErrors()` to the rejected value.

So to submit a form with validation/errors, it would look like:

```js
const submitUser = new Promise((resolve, reject) => {
  fetch('...').then(
    (response) => {
      // do whatever with the response
      resolve(true); // indicate that form is valid
    },
    (errors) => {
      reject(errors); // will set errors of field/form
    });
}

dispatch(actions.submit('user', submitUser));
```

**Comment by davidkpiano** [comment]
@kid These are released in `react-redux-form@0.6.2`, though I still have to write documentation for the new actions. I will close this ticket when the documentation is finished; until then, feel free to play around with the above API and see if it fits your use cases!

**Comment by davidkpiano** [comment]
All the above is now released in `0.7.0`, documentation to follow! 

https://github.com/davidkpiano/react-redux-form/releases/tag/v0.7.0
